src/components/EnterpriseSubsidiesContext/data/hooks.js

```diff
@@ -2,24 +2,53 @@ import { useEffect, useState } from 'react';
 import { logError } from '@edx/frontend-platform/logging';
 import { getConfig } from '@edx/frontend-platform/config';
 import { camelCaseObject } from '@edx/frontend-platform/utils';
+import moment from 'moment';
 
 import EcommerceApiService from '../../../data/services/EcommerceApiService';
+import EnterpriseAccessApiService from '../../../data/services/EnterpriseAccessApiService';
 import LicenseManagerApiService from '../../../data/services/LicenseManagerAPIService';
+import SubsidyApiService from '../../../data/services/EnterpriseSubsidyApiService';
 
-export const useEnterpriseOffers = ({ enablePortalLearnerCreditManagementScreen }) => {
+export const useEnterpriseOffers = ({ enablePortalLearnerCreditManagementScreen, enterpriseId }) => {
   const [offers, setOffers] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [canManageLearnerCredit, setCanManageLearnerCredit] = useState(false);
 
   useEffect(() => {
     const fetchOffers = async () => {
       try {
-        const response = await EcommerceApiService.fetchEnterpriseOffers({
-          isCurrent: true,
-        });
-        const { results } = camelCaseObject(response.data);
-        setOffers(results);
+        const enterpriseAccessResponse = await EnterpriseAccessApiService.getSubsidyAccessPolicies(enterpriseId);
```

Comment:
[clarification] I'm curious why we need to check if there are any policies before retrieving the subsidy metadata. Is there a way to get at the subsidy metadata without needing to call `enterprise-access` first? For example, the subsidy service's `subsidies` API might accept a `enterprise_customer_uuid` query parameter to only return subsidies for the requested enterprise, by passing the need to go through policies to get a subsidy uuid?

Reply:
Excellent point, I'll call the `subsidies` API directly!